test/test_polynomial.cpp

```diff
@@ -298,3 +298,26 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(test_odd_even, T, all_test_types)
     BOOST_CHECK_EQUAL(odd(b), false);
     BOOST_CHECK_EQUAL(even(b), true);
 }
+
+BOOST_AUTO_TEST_CASE_TEMPLATE( test_pow, T, all_test_types )
+{
+    polynomial<T> a(d3a.begin(), d3a.end());
+    polynomial<T> const one = identity_element(std::multiplies< polynomial<T> >());
```

Comment:
Btw, I don't think we have to go to this length to specify one and zero after all unless it is inside a generic algorithm where the binary operation is parameterized. I think when you know the operation it's fine to just say
`polynomial<T> zero;
polynomial<T> one(T(1));`